### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by m7moud**

### CODE REVIEW

File: spec/access-token-introspection/02-access_spec.lua [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        describe("with default config", function()
+            it("responds with unauthorized", function()
+                local r = client:get("/request",
+                                     {headers = {host = "test1.com"}})
+                -- validate that the request fails, response status 401
+                assert.response(r).has.status(401)
+            end)
+        end)
+
+        describe("with require_success = false", function()
```
Comment (vovchynniko) [inline spec/access-token-introspection/02-access_spec.lua:64]: These tests are not enough. We need to test the whole`run` function, not only `require_success` checks.

File: spec/access-token-introspection/01-schema_spec.lua [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    it("validates introspection_map to only have body, header, and static",
+       function()
+        local ok, err = validate({
+            client_id = "CLIENT_ID",
+            client_secret = "CLIENT_SECRET",
+            introspection_endpoint = "http://localhost:8080",
+            introspection_map = {xyz = "zyx"}
+        })
+        assert.falsy(ok)
+        assert.same("unknown field", err.config.introspection_map.xyz)
```
Comment (vovchynniko) [inline spec/access-token-introspection/01-schema_spec.lua:72]: Please test also the know fields: `body`, `header` and `static`.

**Review by vovchynniko — CHANGES_REQUESTED** [review]

File: spec/access-token-introspection/01-schema_spec.lua (hunk 2 above)
Comment (m7moud) [reply spec/access-token-introspection/01-schema_spec.lua:72]: Done.

File: spec/access-token-introspection/02-access_spec.lua (hunk 1 above)
Comment (vovchynniko) [reply spec/access-token-introspection/02-access_spec.lua:64]: I'll create a tech debt JIRA ticket for the future.

**Review by vovchynniko — APPROVED** [review]
Thank you!

File: spec/access-token-introspection/02-access_spec.lua (hunk 1 above)
Comment (vovchynniko) [reply spec/access-token-introspection/02-access_spec.lua:64]: https://medwing.atlassian.net/browse/CSN-532

**Pull request merged**